File: chain-sdk/javascript-closure-angular/API/Client/UniswapV3UniswapDayDataV3DTO.js
goog.provide('API.Client.UniswapV3UniswapDayDataV3DTO');

/**
 * Data accumulated and condensed into day stats for all of Uniswap.
 * @record
 */
API.Client.UniswapV3UniswapDayDataV3DTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.UniswapV3UniswapDayDataV3DTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.UniswapV3UniswapDayDataV3DTO.prototype.recvTime;

/**
 * Number of block in which entity was recorded.
 * @type {!number}
 * @export
 */
API.Client.UniswapV3UniswapDayDataV3DTO.prototype.blockNumber;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.UniswapV3UniswapDayDataV3DTO.prototype.vid;

/**
 * Timestamp rounded to current day by dividing by 86400.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3UniswapDayDataV3DTO.prototype.id;

/**
 * Timestamp rounded to current day by dividing by 86400.
 * @type {!number}
 * @export
 */
API.Client.UniswapV3UniswapDayDataV3DTO.prototype.date;

/**
 * Total volume across all pairs on this day, stored as a derived amount of ETH.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3UniswapDayDataV3DTO.prototype.volumeEth;

/**
 * Total volume across all pairs on this day, stored as a derived amount of USD.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3UniswapDayDataV3DTO.prototype.volumeUsd;

/**
 * Total daily volume in Uniswap derived in terms of USD untracked.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3UniswapDayDataV3DTO.prototype.volumeUsdUntracked;

/**
 * Fees in USD
 * @type {!string}
 * @export
 */
API.Client.UniswapV3UniswapDayDataV3DTO.prototype.feesUsd;

/**
 * @type {!API.Client.NumericsBigInteger}
 * @export
 */
API.Client.UniswapV3UniswapDayDataV3DTO.prototype.txCount;

/**
 * Tvl in terms of USD.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3UniswapDayDataV3DTO.prototype.tvlUsd;

File: chain-sdk/clojure/src/on_chain_api/specs/uniswap_v3_uniswap_day_data_v3_dto.clj
(ns on-chain-api.specs.uniswap-v3-uniswap-day-data-v3-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [on-chain-api.specs.numerics-big-integer :refer :all]
            )
  (:import (java.io File)))


(def uniswap-v3-uniswap-day-data-v3-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :vid) int?
   (ds/opt :id) string?
   (ds/opt :date) int?
   (ds/opt :volume_eth) string?
   (ds/opt :volume_usd) string?
   (ds/opt :volume_usd_untracked) string?
   (ds/opt :fees_usd) string?
   (ds/opt :tx_count) numerics-big-integer-spec
   (ds/opt :tvl_usd) string?
   })

(def uniswap-v3-uniswap-day-data-v3-dto-spec
  (ds/spec
    {:name ::uniswap-v3-uniswap-day-data-v3-dto
     :spec uniswap-v3-uniswap-day-data-v3-dto-data}))
